src/applications/disability-benefits/526EZ/pages/primaryAddress.js

```diff
@@ -392,92 +390,72 @@ const countries = [
   'Zimbabwe'
 ];
 
-export const militaryPostOfficeTypeLabels = { // TODO: determine whether these are necessary
-  APO: 'Army Post Office',
-  FPO: 'Fleet Post Office',
-  DPO: 'Diplomatic Post Office'
-};
-
 const addressUISchema = (addressName, title) => {
   return {
+    'ui:order': [
+      'country',
+      'addressLine1',
+      'addressLine2',
+      'addressLine3',
+      'city',
+      'state',
+      'zipCode'
+    ],
     'ui:title': title,
-    type: {
-      'ui:title': 'Type',
-      'ui:options': {
-        labels: typeLabels
-      }
-    },
     country: {
       'ui:title': 'Country'
     },
-    state: {
-      'ui:title': 'State',
-      'ui:options': {
-        labels: stateLabels,
-        hideIf: formData => {
-          return (
-            _.get(formData, `veteran[${addressName}].country`) !== 'USA'
-          );
-        }
-      }
-    },
     addressLine1: {
-      'ui:title': 'Street'
+      'ui:title': 'Street address'
     },
     addressLine2: {
-      'ui:title': 'Line 2'
+      'ui:title': 'Street address (optional)'
     },
     addressLine3: {
-      'ui:title': 'Line 3'
+      'ui:title': 'Street address (optional)'
     },
     city: {
-      'ui:title': 'City'
+      'ui:title': 'City',
+      'ui:validations': [{
+        options: { addressName },
+        validator: validateMilitaryCity
+      }]
     },
-    militaryStateCode: {
-      'ui:title': 'Military State Code',
+    state: {
+      'ui:title': 'State',
+      'ui:required': ({ veteran }) => (veteran.mailingAddress.country === USA),
       'ui:options': {
         labels: stateLabels,
-        hideIf: formData => _.get(formData, `veteran[${addressName}].type`) !== 'MILITARY'
-      }
+        hideIf: ({ veteran }) => (veteran.mailingAddress.country !== USA),
```

Comment:
State is not required if the address is US, but does that mean it should be hidden? Can states still be provided for international addresses, it seems like these would be helpful?

Reply:
I hid state because that's how we built Letters (country !== USA: hide state field), which is just a so-so reason on its own. But I also learned today that vets360 considers 'state' input for international addresses as invalid, so it seems that we should probably leave it off here as well.

Reply:
Thanks Mark, if we include some sort of state field for international addresses, it looks like they will likely need to be remapped to a province or provinceCode as part of the tranformations and toggle the label on the FE as well. (Accordion to https://github.com/department-of-veterans-affairs/vets.gov-team/issues/10878#issuecomment-396270145)

Reply:
Is including the optional international province field something to check with @goldenmeanie about or should we plan to follow how the common address uiSchema behaves (it includes province for international addresses)?

Reply:
@cehsu, I think go with standard!

Reply:
Sorry, not sure I understand the difference between the two options? To back up for a sec - we never intended to support functionality (e.g., international provinces) that differs from PCIU.
1. We will likely need to populate the state & countries lists from an api at some point - this api will only contain a US states list.
2. Letters went through extensive FE & BE validation for PCIU addresses (note - Letters actually *does* update the user's PCIU address on file), and also loads state / country data from an endpoint. We don't currently support non-US state / province input there.
3. The new Personalization dashboard includes a widget that lets veterans update their address. The specs for this address updater match PCIU / what we did in Letters - only US addresses can have states / provinces.

I understand these things may change with the switch to vets360, but I don't think we should complicate our implementation right now.

Reply:
Thanks for clarifying @markgreenburg, I couldn't tell from the specs that international addresses shouldn't accept states/provinces, I was only aware that they would not be required in that case, also I didn't realize we had already made a design decision on this, so thanks for the context. Let's go with whatever is actually accepted.

Reply:
All good! It's not just you, I think we all struggled with the specs for this initially (Chris made really good specs that are attached to this ticket, but it took us a really long time to get there). Appreciate you taking the time to think critically about whether these are / are not the right changes to make - I'm generally a proponent of using as much common code as possible.